### PR TITLE
feat: mp campaign 0 fees

### PR DESCRIPTION
**Pull request by ChefBingbong**
### Summary
🎁🆕🕒

This pull request adds a feature to the `buyCrypto` state and UI that checks and displays the user's new customer status for the selected on-ramp provider. It also prioritizes MoonPay as an option for new customers and offers them a fee waiver as part of a campaign. The changes affect the files `hooks.ts`, `reducer.ts`, `AccordianItem.tsx`, `usePriceQuoter.ts`, and `actions.ts` in the `buyCrypto` module.

> _Sing, O Muse, of the valiant code warriors who strove_
> _To enhance the `buyCrypto` state with a new action and a flag_
> _That would reveal to them the status of each eager buyer_
> _And grant them special favors from the provider called MoonPay_

### Walkthrough
*  Add a new property `isNewCustomer` to the `buyCrypto` state and a new action `isNewCustomer` to update it ([link](https://github.com/pancakeswap/pancake-frontend/pull/7561/files?diff=unified&w=0#diff-31307291719d5bf9871487cf6df474223e45f0c31d8c834b2006129b8fef0fe5R16), [link](https://github.com/pancakeswap/pancake-frontend/pull/7561/files?diff=unified&w=0#diff-31307291719d5bf9871487cf6df474223e45f0c31d8c834b2006129b8fef0fe5R33), [link](https://github.com/pancakeswap/pancake-frontend/pull/7561/files?diff=unified&w=0#diff-31307291719d5bf9871487cf6df474223e45f0c31d8c834b2006129b8fef0fe5R89), [link](https://github.com/pancakeswap/pancake-frontend/pull/7561/files?diff=unified&w=0#diff-31307291719d5bf9871487cf6df474223e45f0c31d8c834b2006129b8fef0fe5R106), [link](https://github.com/pancakeswap/pancake-frontend/pull/7561/files?diff=unified&w=0#diff-75772925913a4b43e6e0b041906b4443e30f351ceb79d0db3297928e18f1d7c3R9))
*  Add a new parameter `isNewCustomer` to the `usePriceQuotes` hook and a new condition to the `fetchQuotes` function to show the MoonPay provider first for new customers ([link](https://github.com/pancakeswap/pancake-frontend/pull/7561/files?diff=unified&w=0#diff-05cf008d35fa21dfccb294f5fe1bb9a2bc335765a3dab53a36b8b54ff7944e72R19), [link](https://github.com/pancakeswap/pancake-frontend/pull/7561/files?diff=unified&w=0#diff-05cf008d35fa21dfccb294f5fe1bb9a2bc335765a3dab53a36b8b54ff7944e72L56-R62))
*  Add a new variable `address` to the `useBuyCryptoState` hook and a new block of code to the `fetchData` function to fetch the customer details from the on-ramp API and set the `isNewCustomer` property ([link](https://github.com/pancakeswap/pancake-frontend/pull/7561/files?diff=unified&w=0#diff-62400b856d722556576b04bcbb9ff801cfaf142737eec302b955b8ab04a2c1f7R305), [link](https://github.com/pancakeswap/pancake-frontend/pull/7561/files?diff=unified&w=0#diff-62400b856d722556576b04bcbb9ff801cfaf142737eec302b955b8ab04a2c1f7R313-R321), [link](https://github.com/pancakeswap/pancake-frontend/pull/7561/files?diff=unified&w=0#diff-62400b856d722556576b04bcbb9ff801cfaf142737eec302b955b8ab04a2c1f7L322-R337))
*  Add a new import and a new condition to the `isActive` function in the `AccordianItem` component to make the dropdown item active for new customers who are eligible for the MoonPay provider fee campaign ([link](https://github.com/pancakeswap/pancake-frontend/pull/7561/files?diff=unified&w=0#diff-67ab1faec5c6877d56243359d79c8d1f866cc76f69d9e5bef692b20f903ea2f3L12-R15), [link](https://github.com/pancakeswap/pancake-frontend/pull/7561/files?diff=unified&w=0#diff-67ab1faec5c6877d56243359d79c8d1f866cc76f69d9e5bef692b20f903ea2f3L56-R80))
*  Add a new block of code and several new elements to the `AccordianItem` component to render a box with a message, an image, and a timer for the MoonPay provider fee campaign for eligible new customers ([link](https://github.com/pancakeswap/pancake-frontend/pull/7561/files?diff=unified&w=0#diff-67ab1faec5c6877d56243359d79c8d1f866cc76f69d9e5bef692b20f903ea2f3L1-R1), [link](https://github.com/pancakeswap/pancake-frontend/pull/7561/files?diff=unified&w=0#diff-67ab1faec5c6877d56243359d79c8d1f866cc76f69d9e5bef692b20f903ea2f3L19-R39), [link](https://github.com/pancakeswap/pancake-frontend/pull/7561/files?diff=unified&w=0#diff-67ab1faec5c6877d56243359d79c8d1f866cc76f69d9e5bef692b20f903ea2f3L140-R181))

### CODE REVIEW

File: apps/web/src/views/BuyCrypto/components/AccordianDropdown/AccordianItem.tsx [hunk 1]
```diff
@@ -9,13 +9,33 @@ import { useTranslation } from '@pancakeswap/localization'
...
+  const daysLeft = Math.floor(timeDifference / (1000 * 60 * 60 * 24))
+  const hoursLeft = Math.floor((timeDifference % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60))
+  const minutesLeft = Math.floor((timeDifference % (1000 * 60 * 60)) / (1000 * 60))
+
+  return {
+    d: daysLeft,
+    h: hoursLeft,
+    m: minutesLeft,
+  }
+}
```
Comment (ChefMomota) [inline apps/web/src/views/BuyCrypto/components/AccordianDropdown/AccordianItem.tsx:38]: U can use `getTimePeriods`
import getTimePeriods from '@pancakeswap/utils/getTimePeriods'

Comment (ChefBingbong) [reply apps/web/src/views/BuyCrypto/components/AccordianDropdown/AccordianItem.tsx:38]: ohh thanks momota, didnt know this existed. brilliant makes cleaner thank you

**Comment by ChefMomota** [comment]
Screenshot 2023-08-16 at 3 33 03 PM

Is that any place use the `.png` image?

**Comment by ChefBingbong** [comment]
> Screenshot 2023-08-16 at 3 33 03 PM
> 
> Is that any place use the `.png` image?

no longer. i switched use out for svg. will remove

**Pull request closed without merging**